### DISCUSSION TITLE
Silent Uninstall Option for NVM for Windows?

### DISCUSSION THREAD

**Discussion by phoenix14830** [coreybutler/nvm-windows#879] [Support] [answered]
The company I work for wants all applications in Intune (Microsoft Endpoint Manager.) There is a "verysilent" switch for the installer of NVM for Windows, but I can't find a way to do the uninstaller silently, which is required for Intune packages. Any popup windows will hang the package indefinitely until it eventually times out and gives a cryptic error about why the process failed. 

Are there uninstall options for this application and if so, is there an option for silent uninstall? 
If not, can that be added as an upgrade to the uninstaller?

Edit:
As a workaround, I made a script that calls the uninstaller, waits a few seconds, then calls this:
Start-Sleep -Seconds 3 
$popupName = 'Uninstall'
$popup = New-Object -ComObject wscript.shell
if ($popup.AppActivate($popupName)) 
{
    $popup.SendKeys('{Enter}')
}

**Comment by coreybutler** [comment, marked as answer]
I've never tried a verysilent install, but it should be possible.

There isn't a specific silent uninstaller. If you're preparing a package, you may be able to silence the prompts in the uninstaller though. Alternatively, the manual uninstallation isn't too complex. There's a guide for how the install/uninstall works at https://github.com/coreybutler/nvm-windows/wiki/Hacking.

As for new features, NVM4W is in a feature freeze while I work on [Runtime, the successor project](https://github.com/coreybutler/nvm-windows/wiki/Runtime). Runtime is being built with enterprise/company needs in mind. For anyone who wants to use NVM4W/Runtime in a company environment, you can sign up to be notified when the Runtime beta is ready by completing [this form](https://t.co/oGqQCM9FPx) (the last survey question allows you to sign up). I'll also post Runtime updates on Twitter (assuming it's still free): [@goldglovecb](https://twitter.com/goldglovecb).
